### ISSUE TITLE
Databricks dialect doesn't support named parameters

### ISSUE THREAD

**Issue by TheCleric** [sqlfluff/sqlfluff#5944]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When using the named parameter syntax while calling a function in Databricks SQL an error occurs when parsing/linting/fixing.

### Expected Behaviour

The sql should parse/lint/fix correctly when using named parameters as documented in https://docs.databricks.com/en/sql/language-manual/sql-ref-function-invocation.html#named-parameter-invocation

### Observed Behaviour

When parsing a sql file using named parameters, an error like the following occurs:

`Found unparsable section: "=> 'csv',\n  header => true"`

### How to reproduce

Here's an example sql command with this issue:

```sql
CREATE OR REFRESH LIVE TABLE location AS
SELECT * FROM READ_FILES(
  's3://bucket/location.csv',
  format => 'csv',
  header => true);
```

### Dialect

Databricks

### Version

sqlfluff, version 3.0.7
Python 3.10.12

### Configuration

```cfg
[sqlfluff]
dialect = databricks
templater = placeholder
max_line_length = 120
exclude_rules = CP02
large_file_skip_byte_limit = 0

[sqlfluff:indentation]
tab_space_size = 2
indented_joins = True
indented_ctes = True
indented_using_on = True
indented_on_contents = True
# indented_then = True
# indented_then_contents = True
allow_implicit_indents = True
template_blocks_indent = True

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:templater:placeholder]
param_regex = \${(?P<param_name>[-\w.]+)}
ese-pipeline.catalog = ese_dev
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by TheCleric** [comment]
I suspect the fix for this will be similar to the fix in https://github.com/sqlfluff/sqlfluff/pull/5027.
